From 6f231353afa3b486b5d285e4187ce4e1db377f52 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Tue, 17 Feb 2026 16:19:21 +0000
Subject: [PATCH 2/4] Include APN configuration database in build

The treble build inherits aosp_base_telephony but does not include the
AOSP APN database. The apns-full-conf.xml module has no_full_install
set in its Android.bp, so it is only installed when explicitly pulled
in. The GSI Android.bp filesystem target includes it, but the treble
makefile-based build does not use that target.

Copy apns-full-conf.xml to the product partition as apns-conf.xml,
matching how device/generic/common/gsi_product.mk does it. Without
this, TelephonyProvider has no carrier APN entries and mobile data
cannot be configured automatically.

Change-Id: I0000000000000000000000000000000000000016
---
 base.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/base.mk b/base.mk
index 2f9a77a..f59cae3 100644
--- a/base.mk
+++ b/base.mk
@@ -24,6 +24,10 @@ $(call inherit-product, vendor/hardware_overlay/overlay.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/core_64_bit.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_base_telephony.mk)
 
+# APN configuration database
+PRODUCT_COPY_FILES += \
+    device/sample/etc/apns-full-conf.xml:$(TARGET_COPY_OUT_PRODUCT)/etc/apns-conf.xml
+
 # Those overrides are here because Huawei's init read properties
 # from /system/etc/prop.default, then /vendor/build.prop, then /system/build.prop
 # So we need to set our props in prop.default
-- 
2.52.0

